src/main/java/ch/njol/skript/lang/SyntaxElement.java

```diff
@@ -12,7 +12,19 @@
 public interface SyntaxElement {
 
 	/**
-	 * Called just after the constructor.
+	 * Called immediately after the constructor. This should be used to do any work that need to be done prior to
+	 * downstream initialization. This is not intended to be used by syntaxes directly, but by parent classes to do
+	 * work prior to the initialization of the child classes.
+	 *
+	 * @return Whether this expression was pre-initialised successfully.
+	 * 			An error should be printed prior to returning false to specify the cause.
+	 */
+	default boolean preInit() {
```

Comment:
maybe this could be marked as `@Internal` for extra clarity

Reply:
I don't think it should be internal, as add-ons could get a lot of use out of it.